src/Aspire.Hosting/ApplicationModel/AspireStore.cs

```diff
@@ -29,56 +30,52 @@ public AspireStore(string basePath)
 
     public string BasePath => _basePath;
 
-    public string GetFileNameWithContent(string filenameTemplate, string sourceFilename)
+    public string GetFileNameWithContent(string filenameTemplate, Stream contentStream)
     {
         ArgumentException.ThrowIfNullOrWhiteSpace(filenameTemplate);
-        ArgumentException.ThrowIfNullOrWhiteSpace(sourceFilename);
-
-        if (!File.Exists(sourceFilename))
-        {
-            throw new FileNotFoundException("The source file does not exist.", sourceFilename);
-        }
+        ArgumentNullException.ThrowIfNull(contentStream);
 
         EnsureDirectory();
 
         // Strip any folder information from the filename.
         filenameTemplate = Path.GetFileName(filenameTemplate);
 
-        var hashStream = File.OpenRead(sourceFilename);
+        // Create a temporary file to write the content to.
+        var tempFileName = Path.GetTempFileName();
 
-        // Compute the hash of the content.
-        var hash = SHA256.HashData(hashStream);
+        // Fast, non-cryptographic hash.
+        var hash = new XxHash3();
 
-        hashStream.Dispose();
+        // Write the content to the temporary file while also building a hash.
+        using (var fileStream = File.OpenWrite(tempFileName))
```

Comment:
You can do this without buffering yourself if you implement a new stream class that hashes while also forwarding to the file.

See https://github.com/dotnet/sdk/blob/5c5a92beffab32631feb44515188c298b53598d2/src/Containers/Microsoft.NET.Build.Containers/Layer.cs#L223 for an example.